### PR TITLE
release: 1.21

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: lots
Docs Changes: indeed
Release Notes: also yes.

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -15,14 +15,13 @@ Minor Behavior Changes
...
 * json: switching from rapidjson to nlohmann/json. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.remove_legacy_json`` to false.
 * listener: destroy per network filter chain stats when a network filter chain is removed during the listener in place update.
-* quic: add back the support for IETF draft 29 which is guarded via ``envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_disable_version_draft_29``. It is off by default so Envoy only supports RFCv1 without flipping this runtime guard explicitly. Draft 29 is not recommended for use.
```
Comment (alyssawilk) [inline docs/root/version_history/current.rst:24]: Took this out becasue quic doesn't need release notes yet, and advertising something we don't want folks to use seems silly.

**Comment by alyssawilk** [comment]
please to ignore coverage failure -  I'm sorting that out elsewhre

docs page here for your convenience:
https://storage.googleapis.com/envoy-pr/ac3f373/docs/version_history/current.html

File: docs/root/version_history/current.rst [hunk 2]
```diff
@@ -15,14 +15,13 @@ Minor Behavior Changes
...
-* router: take elapsed time into account when setting the x-envoy-expected-rq-timeout-ms header for retries, and never send a value that's longer than the request timeout. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.update_expected_rq_timeout_on_retry`` to false.
```
Comment (adisuissa) [inline docs/root/version_history/current.rst:25]: Why was this removed?

**Review by adisuissa — COMMENTED** [review]
Thanks for managing the release and cleaning up the docs.
Left one question.

File: docs/root/version_history/current.rst [hunk 3]
```diff
@@ -15,14 +15,13 @@ Minor Behavior Changes
...
+* router: take elapsed time into account when setting the 11x-envoy-expected-rq-timeout-ms header11 for retries, and never send a value that's longer than the request timeout. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.update_expected_rq_timeout_on_retry`` to false.
 * stream_info: response code details with empty space characters (' ', '\t', '\f', '\v', '\n', '\r') is not accepted by the ``setResponseCodeDetails()`` API.
```
Comment (adisuissa) [inline docs/root/version_history/current.rst:25]: The backslashes do not appear in the rendered doc (https://storage.googleapis.com/envoy-pr/ac3f373/docs/version_history/current.html).
Should this be `\\`?

```suggestion
* stream_info: response code details with empty space characters (' ', '\\t', '\\f', '\\v', '\\n', '\\r') is not accepted by the ``setResponseCodeDetails()`` API.
```

File: RELEASES.md [hunk 4]
```diff
@@ -75,7 +75,7 @@ As of 2022, the targeted release dates are the 15th day of each quarter.
 | 1.17.0  | 2020/12/31 | 2021/01/11 |  +11 days  | 2022/01/11  |
 | 1.18.0  | 2021/03/31 | 2021/04/15 |  +15 days  | 2022/04/15  |
 | 1.19.0  | 2021/06/30 | 2021/07/13 |  +13 days  | 2022/07/13  |
-| 1.20.0  | 2021/09/30 | 2021/10/05 |   +5 days  | 2022/10/13  |
-| 1.21.0  | 2022/01/15 |            |            |             |
+| 1.20.0  | 2021/09/30 | 2021/10/05 |   +5 days  | 2022/10/05  |
+| 1.21.0  | 2022/01/15 | 2022/01/12 |   -3 days  | 2023/01/12  |
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply RELEASES.md:79]: yeah I think we add after the release is cut and we _definitely_ want a release manager for Q1 - waiting for folks to get back to us sadly.

**Comment by alyssawilk** [comment]
it's getting there!
latest version here: https://storage.googleapis.com/envoy-pr/2043334/docs/version_history/current.html
if @envoyproxy/envoy-maintainers could take a look, many reviews make for few typos :-P

**Review by adisuissa — COMMENTED** [review]
Looked at it again, LGTM.
2 nits (sorry again for commenting like this again, but cannot get github UI to add comments on these lines):

Line 85:
Before: `* http: added support for %REQUESTED_SERVER_NAME% to extract SNI as a custom header.`
After: `* http: added support for ``%REQUESTED_SERVER_NAME%`` to extract SNI as a custom header.`

Line 86:
Before: `* http: added support for %VIRTUAL_CLUSTER_NAME% to extract the matched Virtual Cluster name as a custom header.`
After: `* http: added support for ``%VIRTUAL_CLUSTER_NAME%`` to extract the matched Virtual Cluster name as a custom header.`

File: docs/root/version_history/current.rst [hunk 5]
```diff
@@ -14,18 +14,17 @@ Minor Behavior Changes
...
 * listener: destroy per network filter chain stats when a network filter chain is removed during the listener in place update.
-* quic: add back the support for IETF draft 29 which is guarded via ``envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_disable_version_draft_29``. It is off by default so Envoy only supports RFCv1 without flipping this runtime guard explicitly. Draft 29 is not recommended for use.
-* router: take elapsed time into account when setting the x-envoy-expected-rq-timeout-ms header for retries, and never send a value that's longer than the request timeout. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.update_expected_rq_timeout_on_retry`` to false.
+* router: take elapsed time into account when setting the ``x-envoy-expected-rq-timeout-ms header`` for retries, and never send a value that's longer than the request timeout. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.update_expected_rq_timeout_on_retry`` to false.
 * stateful session http filter: added :ref:`stateful session http filter <config_http_filters_stateful_session>`.
```
Comment (adisuissa) [inline docs/root/version_history/current.rst:26]: Should this be under `New Features`?

**Review by adisuissa — COMMENTED** [review]
One more comments from latest main merge :)

**Comment by alyssawilk** [comment]
Final docs here: https://storage.googleapis.com/envoy-pr/93d993e/docs/version_history/current.html

**Review by adisuissa — APPROVED** [review]
LGTM, thanks @alyssawilk!
